lib/rubocop/cop/style/class_vars.rb

```diff
@@ -38,10 +38,23 @@ def on_cvasgn(node)
           add_offense(node, location: :name)
         end
 
+        def on_send(node)
+          return unless class_var_set?(node)
+
+          _, _, name, = *node
+          add_offense(name)
+        end
```

Comment:
It seems to be simple as follows.

```diff
diff --git a/lib/rubocop/cop/style/class_vars.rb b/lib/rubocop/cop/style/class_vars.rb
index e4772213e..e50bc5506 100644
--- a/lib/rubocop/cop/style/class_vars.rb
+++ b/lib/rubocop/cop/style/class_vars.rb
@@ -39,22 +39,15 @@ module RuboCop
         end

         def on_send(node)
-          return unless class_var_set?(node)
+          return unless node.method?(:class_variable_set)

-          _, _, name, = *node
-          add_offense(name)
+          add_offense(node.arguments.first)
         end

         def message(node)
           class_var, = *node
           format(MSG, class_var: class_var)
         end
-
-        private
-
-        def_node_matcher :class_var_set?, <<~PATTERN
-          (send _ :class_variable_set _ _)
-        PATTERN
       end
     end
   end
```

Reply:
Nice. Still getting to grips with the different ways of matching nodes. Thanks